### PR TITLE
Improve envoyfilter patch implementation

### PR DESCRIPTION
**Pull request by wulianglongrd**
I think these changes help 1. Improve readability 2. Enhance robustness (finding the position and replacing or inserting is not an easy task, and LDS/RDS/CDS all have similar logic)

Currently, only part of it has been modified. Once this PR is merged, I will modify other places in the next PR.

### CODE REVIEW

File: pkg/slices/slices.go [hunk 1]
```diff
@@ -238,3 +238,74 @@ func GroupUnique[T any, K comparable](data []T, f func(T) K) map[K]T {
...
+	s = append(s, toInsert) // for grow the cap
+	copy(s[idx+1:], s[idx:])
+	s[idx] = toInsert
+
+	return s, true
+}
+
+// InsertAfterFunc find and insert an element after the found element.
+// If the f function returns true, the returned value will be inserted after the found element.
+func InsertAfterFunc[E any](s []E, f func(e E) (bool, E)) ([]E, bool) {
```
Comment (howardjohn) [inline pkg/slices/slices.go:282]: nit: maybe put this in envoyfilter package? Its very very niche and unlikely to be used outside of that package, and is a bit confusing as a general purpose helper

Comment (howardjohn) [reply pkg/slices/slices.go:282]: (same with insertbefore)

**Review by howardjohn — COMMENTED** [review]
LGTM

**Review by ramaraochavali — APPROVED** [review]

File: pkg/slices/slices.go (hunk 1 above)
Comment (wulianglongrd) [reply pkg/slices/slices.go]: Thanks, updated.

**Pull request merged**